src/org/labkey/test/tests/list/ListTest.java

```diff
@@ -683,6 +688,91 @@ public void testBatchInsertErrors() throws IOException, CommandException
+                .withColumns(cols);
+        DomainResponse createResponse = dgen.createList(createDefaultConnection(true), "key");
+        Domain listDomain = createResponse.getDomain();
+        List<PropertyDescriptor> listFields = createResponse.getDomain().getFields();
+        listFields.add(new FieldDefinition("volume", FieldDefinition.ColumnType.Decimal));
+        listDomain.setFields(listFields);
+
+        // now save with an extra field
+        SaveDomainCommand saveCmd = new SaveDomainCommand(info.getSchema(), info.getTable());
+        saveCmd.setDomainDesign(listDomain);
+        DomainResponse saveResponse = saveCmd.execute(createDefaultConnection(true), info.getFolder());
+
+        // now verify
+        assertEquals(listFields.size(), saveResponse.getDomain().getFields().size());
+        for (PropertyDescriptor expectedField : listFields)
+        {
+            checker().verifyTrue( "expect field [" + expectedField.getName() + "] with type [" +expectedField.getRangeURI()+ "]",
+                    saveResponse.getDomain().getFields().stream()
+                    .anyMatch(a -> a.getName().equals(expectedField.getName()) &&
+                            a.getRangeURI().endsWith(expectedField.getRangeURI())));
+        }
+    }
+
+    @Test
+    public void testRemoveColumnOverAPI() throws Exception
+    {
+        List<FieldDefinition> cols = Arrays.asList(
+                new FieldDefinition("name", FieldDefinition.ColumnType.String),
+                new FieldDefinition("title", FieldDefinition.ColumnType.String),
+                new FieldDefinition("dewey", FieldDefinition.ColumnType.Decimal),
+                new FieldDefinition("removeMe", FieldDefinition.ColumnType.Decimal)
+        );
+        String listName = "remoteApiListTestRemoveColumn";
+        FieldDefinition.LookupInfo info = new FieldDefinition.LookupInfo(getProjectName(), "lists", listName);
+        TestDataGenerator dgen = new TestDataGenerator(info)
+                .withColumns(cols);
+        DomainResponse createResponse = dgen.createList(createDefaultConnection(true), "key");
+        Domain listDomain = createResponse.getDomain();
+        List<PropertyDescriptor> listFields = createResponse.getDomain().getFields();
+        listFields.removeIf(a-> a.getName().equals("removeMe"));
+        listDomain.setFields(listFields);
+
+        SaveDomainCommand saveCmd = new SaveDomainCommand(info.getSchema(), info.getTable());
+        saveCmd.setDomainDesign(listDomain);
+        DomainResponse saveResponse = saveCmd.execute(createDefaultConnection(true), info.getFolder());
+
+        checker().verifyFalse("expect field [title] with type [" +FieldDefinition.ColumnType.String.getConceptURI()+ "]",
+                saveResponse.getDomain().getFields().stream()
+                        .anyMatch(a -> a.getName().equals("removeMe") &&
+                                a.getRangeURI().endsWith(FieldDefinition.ColumnType.String.getConceptURI())));
```

Comment:
Why are you using `conceptURI` here? From my experimentation, it is just to provide UI styling to particular types. e.g. A 'Sample' column is really just a lookup to `exp.Materials` but the 'sample' concept makes the domain designer look different.

That being said; still having a "removeMe" column, just with a different type seems like a failure.

Reply:
it was the difference between "float" and "http://www.labkey.org/exp/xml#float", and I think the API is now regarding the full URL as the canonical way of describing the rangeURI even though it will accept "float" as an acceptable input
I just thought 'conceptURI' was odd nomenclature for a field that holds the value for a canonical rangeURI